jhipster: add page (#4047)
tldr-pages/tldr: pages/common/jhipster.md

# jhipster

> > Generates a monolithic or microservices project after prompting user.

Reviewer: Generator for creating Spring and/or Angular/React/Vue projects.
```suggestion
> Generator from creating Spring and/or Angular/React/Vue projects.
```

Author: I disagree with this suggestion as you can also generate a backend with Kotlin, .NET, NodeJS or Quarkus so I'd rather prefer to stay generic in the description if you don't mind?

Reviewer: Hi @fdelbrayelle 

No problem, as I have no experience with JHipster my suggestion was based on the official [tech stack](https://www.jhipster.tech/tech-stack/)

Even so I think we could narrow a little bit the description, maybe focusing on the web scenario.

What do you think about "Web application generator using either monolithic or microservice architecture"

Author: It sounds not bad at all!

> More information: <https://www.jhipster.tech/>.

- Generate a simple full-stack project (monolithic or microservices):

`jhipster`

> - Generate a simple frontend project with `--skip-server` or a simple backend project with `--skip-client`:

Reviewer 2: This is still kinda problematic, as it's a mesh of two examples

Author: I've splitted them again but removed the one about `docker-compose`

> - Generate a simple frontend project (with OPTION = --skip-server) or a simple backend project (with OPTION = --skip-client):

Reviewer 2: These two OPTIONs can be split in two examples. Since then there would be too many examples in this page, maybe you want to move some of the subcommands into their respective pages? (e.g. `jhpister-upgrade` can have example from that subcommand)

Author: As `jhipster upgrade` [has no specific option](https://www.jhipster.tech/upgrading-an-application/), I'd rather prefer to let it that way if it's not a problem :)

> - Generate a simple frontend project (with OPTION = --skip-server) or a simple backend project (with OPTION = --skip-client):
> 
> `jhipster [OPTION]`

Reviewer 3: ```suggestion
- Generate a simple frontend project with `--skip-server` or a simple backend project with `--skip-client`:

`jhipster {{option}}`
```
It might be simple and clear (although I would wait for the review by another one).

> - Upgrade a generated project with the last JHipster updates:

Author: Hi @andrik, thank you for your suggestion!

`jhipster upgrade`

- Add a new entity to a generated project:

`jhipster entity {{entity_name}}`

> - Import a JDL file to configure your application (entities...) - See: https://start.jhipster.tech/jdl-studio/:

Reviewer 2: ```suggestion
- Import a JDL file to configure your applications (see: https://start.jhipster.tech/jdl-studio/):
```

> - Import a JDL file to configure your application (entities...) - See: https://start.jhipster.tech/jdl-studio/:

Reviewer 3: ```suggestion
- Import a JDL file to configure your application:
```

I think it should not be a link in the command description, and the parentheses may be unnecessary.

> `jhipster import-jdl {{first_file.jh}} {{second_file.jh}}... {{n_file.jh}}`

Reviewer 3: I have doubts with the usage of the `...` in this example, but I don't remember how we should address this, so it is possible another mate might help with this.

Author: Yep I've searched about it in man pages and it seems it's formatted that way:

```
ls [OPTION]... [FILE]...
```

Reviewer 2: ```suggestion
`jhipster import-jdl {{first_file.jh second_file.jh ... n_file.jh}}`
```
Maybe like this, but not sure.

Author: Yep it seems it's like it by searching on the other pages with `\{.*\.\.\.\}`

- Generate a CI/CD pipeline for your application:

`jhipster ci-cd`

> - Generate a docker-compose.yml file for your application(s):

Reviewer 2: ```suggestion
- Generate a docker-compose.yml file for your applications:
```

`jhipster docker-compose`

> - Generate a Kubernetes configuration (kubectl-apply.sh and related yml files) for your application(s):

Reviewer 2: ```suggestion
- Generate a Kubernetes configuration for your applications:
```
